x/lscosmos/keeper/delegation_strategy.go

```diff
@@ -168,8 +168,14 @@ func distributeCoinsAmongstValSet(ws types.WeightedAddressAmounts, coin sdk.Coin
 			valAddrAmts = append(valAddrAmts, types.ValAddressAmount{ValidatorAddr: w.Address, Amount: coin})
 			return valAddrAmts, sdk.NewInt64Coin(coin.Denom, 0)
 		}
-		valAddrAmts = append(valAddrAmts, types.ValAddressAmount{ValidatorAddr: w.Address, Amount: w.Coin()})
-		coin = coin.SubAmount(w.Amount)
+
+		tempAmount := w.Amount
+		if w.Amount.IsNegative() {
+			tempAmount = sdk.ZeroInt()
```

Comment:
i think negatives can be returned early..

Reply:
I don't see the need in the test cases I have added.

Reply:
not need, just small optimisation